xdsl/frontend/pyast/context.py

```diff
@@ -125,13 +125,19 @@ def decorator(func: Callable[P, R]) -> PyASTProgram[P, R]:
             func_file, func_globals, func_ast = self._get_func_info(
                 currentframe(), func, decorated_func
             )
+
+            # Override default post transforms if desymref is unset
+            kwargs: dict[str, Any] = {}
+            if not desymref:
+                kwargs["post_transforms"] = []
```

Comment:
it should pass `self.post_transforms` here, as users should set those on the ctx directly